src/LuaTransformer.ts

```diff
@@ -3186,7 +3228,17 @@ export class LuaTransformer {
             return tstl.createCallExpression(this.transformExpression(node.expression), parameters);
         } else {
             // Replace last . with : here
-            const name = node.expression.name.escapedText;
+            let name = node.expression.name.escapedText as string;
+
+            // catch usage of Lua keywords
+            if (this.luaKeywords.has(name)) {
+              if ( this.translateKeywords.has(name) ) {
```

Comment:
It should be done only when `ownerType` is a promise, because in future (#429) it might be valid to use keywords as property names

Reply:
The promises in the es6 standard explicitly allow arbitrary objects with "then" method to be handled as a PromiseLike. 

#429 suggests  `const then = 1` -> `local ____then = 1`. I chose __TS__then instead.
As soon as #429 is there my current translation to __TS__then is obsolete.

Reply:
In #429 `____then` would be used only for identifiers. In this case it would be `promise["then"]`. Of course that works, but I see a point in using some renamed property because of performance. `promise:__TS__then(...)` would be better than `promise["then"](promise, ...)`, because `promise` may be non-pure value, and thus be compiled to iife.